www/generate_benchmark_means.js

```diff
@@ -0,0 +1,109 @@
+// Copyright 2016 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License.
+//
+// Author: William Haack (will@cockroachlabs.com)
+
+function loadGeometricMeanListFromJSON(callback) {
+    var xobj = new XMLHttpRequest();
+    xobj.overrideMimeType('application/json');
+    xobj.open('GET', 'geometric_means.json', true);
+    xobj.onreadystatechange = function () {
+	if (xobj.readyState == 4 && xobj.status == '200') {
+	    callback(xobj.responseText);
+	}
+    };
+    xobj.send(null);
+}
+
+function getUrlVars() {
+    var vars = [], hash;
+    var hashes =window.location.href.slice(
+	window.location.href.indexOf('?') + 1).split('&');
+    for(var i = 0; i < hashes.length; i++)
+    {
+	hash = hashes[i].split('=');
+	vars.push(hash[0]);
+	vars[hash[0]] = hash[1];
+    }
+    return vars;
+}
+
+function displayBenchMeans(dirName, dirNameToMeanList) {
+    var dateToAverages = new Array();
+    // dateToAverages is an array of arrays so we can sort the data by date.
+    for (var i in dirNameToMeanList[dirName]) {
+	meanData = dirNameToMeanList[dirName][i];
+	dateToAverages.push([meanData['Date'],
+			     meanData['NMean'],
+			     meanData['AMean'],
+			     meanData['BMean'],
+			     meanData['MMean']]);
+    }
+    dateToAverages.sort(function(a, b) {
+	return sortDates(a[0], b[0]);
+    });
+    var dates = [];
+    var NMean = [];
+    var AMean = [];
+    var BMean = [];
+    var MMean = [];
+    for (var i in dateToAverages) {
+	meanData = dateToAverages[i];
+	console.log(meanData[0]);
+	dates.push(meanData[0]);
+	NMean.push(meanData[1]);
+	AMean.push(meanData[2]);
+	BMean.push(meanData[3]);
+	MMean.push(meanData[4]);
+    }
```

Comment:
I feel like you could do this in one pass. Could you sort `dirNameToMeanList[dirName]` by date first, then map NMean, AMean, etc to arrays?